### PR TITLE
SCP issue fixes: setting the timestamp of directory and enable run_err to report error back to client

### PR DESCRIPTION
**Pull request by bingbing8**
https://github.com/PowerShell/Win32-OpenSSH/issues/465
https://github.com/PowerShell/Win32-OpenSSH/issues/464

### CODE REVIEW

File: contrib/win32/win32compat/misc.c [hunk 1]
```diff
@@ -455,20 +455,55 @@ w32_chown(const char *pathname, unsigned int owner, unsigned int group) {
 	return -1;
 }
 
+void
+UnixTimeToFileTime(ULONG t, LPFILETIME pft) {
```
Comment (manojampalam) [inline contrib/win32/win32compat/misc.c]: need to follow rest of codes naming convention of routine names (all lower case, words separated by "_"). 
this looks like a local routine, make it "static"

File: contrib/win32/win32compat/misc.c [hunk 2]
```diff
@@ -455,20 +455,55 @@ w32_chown(const char *pathname, unsigned int owner, unsigned int group) {
...
+	
+	ULONGLONG ull;
+	ull = UInt32x32To64(t, 10000000) + 116444736000000000;
+
+	pft->dwLowDateTime = (DWORD)ull;
+	pft->dwHighDateTime = (DWORD)(ull >> 32);
+}
+
+int
+w32_settimes(wchar_t * path, FILETIME *cretime, FILETIME *acttime, FILETIME *modtime) {
```
Comment (manojampalam) [inline contrib/win32/win32compat/misc.c]: remove w32 prefix and make it "static"

**Review by manojampalam — COMMENTED** [review]
please follow up

**Comment by bingbing8** [comment]
close and reopen to triger CI build.

File: contrib/win32/win32compat/misc.c [hunk 3]
```diff
@@ -455,20 +455,55 @@ w32_chown(const char *pathname, unsigned int owner, unsigned int group) {
...
+	FILETIME acttime, modtime;
 	wchar_t *resolvedPathName_utf16 = utf8_to_utf16(sanitized_path(filename));
 	if (resolvedPathName_utf16 == NULL) {
 		errno = ENOMEM;
 		return -1;
 	}
-	/* _utime only set time for filename; TODO - need a different logic to set times for directory*/
-	ret = _wutime(resolvedPathName_utf16, &ub);
+	memset(&acttime, 0, sizeof(FILETIME));
+	memset(&modtime, 0, sizeof(FILETIME));
```
Comment (manojampalam) [inline contrib/win32/win32compat/misc.c]: sample comment1

File: contrib/win32/win32compat/misc.c [hunk 4]
```diff
@@ -572,11 +607,16 @@ w32_mkdir(const char *path_utf8, unsigned short mode) {
 	}
 	int returnStatus = _wmkdir(path_utf16);
 	if (returnStatus < 0) {
+		free(path_utf16);
 		return -1;
 	}
-	returnStatus = _wchmod(path_utf16, mode);
+
```
Comment (manojampalam) [inline contrib/win32/win32compat/misc.c]: sample comment2

File: contrib/win32/win32compat/misc.c (hunk 3 above)
Comment (bingbing8) [inline contrib/win32/win32compat/misc.c]: response sample

File: contrib/win32/win32compat/misc.c [hunk 5]
```diff
@@ -572,11 +607,16 @@ w32_mkdir(const char *path_utf8, unsigned short mode) {
...
 	}
-	returnStatus = _wchmod(path_utf16, mode);
+	
+	mode_t curmask = _umask(0);
+	_umask(curmask);
+	
+	returnStatus = _wchmod(path_utf16, mode & ~curmask & (_S_IREAD | _S_IWRITE));
 	free(path_utf16);
-	/*TODO: map mode*/
+
```
Comment (manojampalam) [inline contrib/win32/win32compat/misc.c]: sample comment3

File: contrib/win32/win32compat/misc.c [hunk 6]
```diff
@@ -455,20 +455,55 @@ w32_chown(const char *pathname, unsigned int owner, unsigned int group) {
 	return -1;
 }
 
+static void
```
Comment (manojampalam) [inline contrib/win32/win32compat/misc.c]: comment 1

File: contrib/win32/win32compat/misc.c [hunk 7]
```diff
@@ -455,20 +455,55 @@ w32_chown(const char *pathname, unsigned int owner, unsigned int group) {
...
+		NULL, OPEN_EXISTING, FILE_FLAG_BACKUP_SEMANTICS, NULL);
+
+	if (handle == INVALID_HANDLE_VALUE) {
+		errno = GetLastError;
+		debug("w32_settimes - CreateFileW ERROR:%d", errno);
+		return -1;
+	}
+
+	if (SetFileTime(handle, cretime, acttime, modtime) == 0)
+	{
```
Comment (manojampalam) [inline contrib/win32/win32compat/misc.c]: comment 2

**Pull request merged**